app/models/core_induction_programme.rb

```diff
@@ -1,15 +1,14 @@
 # frozen_string_literal: true
 
 class CoreInductionProgramme < ApplicationRecord
-  has_many :course_years, dependent: :delete_all
   belongs_to :course_year_one, class_name: "CourseYear", optional: true
   belongs_to :course_year_two, class_name: "CourseYear", optional: true
   has_many :early_career_teacher_profiles
   has_many :early_career_teachers, through: :early_career_teacher_profiles, source: :user
   has_many :mentor_materials
 
   def course_years
-    [course_year_one, course_year_two].filter(&:present?)
```

Comment:
I've changed this so it returns an ActiveRecord thing so we can chain things if needed. also prevents the 2 initial queries